File: Ruby/CardCounting.java
public class CardCounting
{
	public static int operations = 0;
	public static void main(String[] args)
	{
		System.out.println(factorial(5));
		System.out.println(operations);
	}
	public static int factorial(int x)
	{
		if(x <= 1)
		{
			operations++;
			return 1;
		}else{
			operations++;
			return factorial(x-1) * x;
		}
	}
}

File: Ruby/cardcounting.rb


def main
	cCounter = CardCounter.new()
	puts cCounter.factorial(5)
	puts cCounter.operations
end

class CardCounter
	attr_accessor :operations
	def initialize()
		@operations = 0
	end
	def factorial(x)
		if(x <= 1)
			@operations += 1
			1
		else
			@operations += 1
			factorial(x-1) * x
		end
	end
end

main